tasks/pkg-windows.yml

```
@@ -106,4 +106,4 @@
   win_file:
     path: "{{ download_msi_result.dest }}"
     state: absent
-  when: (not datadog_skip_windows_install) and (download_msi_result.status_code == 200)
+  when: (not datadog_skip_windows_install) and (download_msi_result.status_code == 200) and (not ansible_check_mode)
```

```suggestion
  when: (not datadog_skip_windows_install) and (not ansible_check_mode) and (download_msi_result.status_code == 200)
```

The `(not ansible_check_mode)` condition needs to be present *before* `(download_msi_result.status_code == 200)`. That way, Ansible won't try to access `download_msi_result` variable in check mode, thanks to lazy evaluation of conditions (with the way you did it, this would still fail in check mode, because Ansible would still attempt to access the `download_msi_result` variable).